tests/unit/test_index.cpp

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright 2018-2019 Redis Labs Ltd. and Contributors
+ *
+ * This file is available under the Apache License, Version 2.0,
+ * modified with the Commons Clause restriction.
+ */
+
+#include "../../deps/googletest/include/gtest/gtest.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "../../src/index/index.h"
+
+#ifdef __cplusplus
+}
+#endif
+
+class IndexTest: public ::testing::Test {
+  protected:
+    static const size_t expected_n;
+    static const char *label;
+    static char *str_key;
+    static char *num_key;
+    static int label_id;
+    static Graph *g;
+    static Index *str_idx;
+    static Index *num_idx;
+
+    static void SetUpTestCase() {
+      srand(time(NULL));
+    }
+
+    static void TearDownTestCase() {
+      Index_Free(str_idx);
+      Index_Free(num_idx);
+
+      Graph_Free(g);
+    }
+
+    static Index* _indexCreate(Graph *g, const char *label, const char *property) {
+      const GrB_Matrix label_matrix = Graph_GetLabelMatrix(g, label_id);
+      return buildIndex(g, label_matrix, label, property);
+    }
+
+    static Graph* build_test_graph() {
+      // Allocate graph
+      size_t n = expected_n;
+      Graph *g = Graph_New(n);
+
+      // Build a label matrix and add to graph
+      // (this does not need to be associated with an actual label string)
+      label_id = Graph_AddLabelMatrix(g);
+
+      // Associate all nodes in graph with this label
+      int label_ids[n];
+      for (int i = 0; i < n; i ++) {
+        label_ids[i] = label_id;
+      }
+
+      // Populate graph with nodes
+      NodeIterator *it;
+      Graph_CreateNodes(g, n, label_ids, &it);
+
+      // Variables to store data for node properties
+      Node *node;
+      char *prop_keys[2];
+      prop_keys[0] = str_key;
+      prop_keys[1] = num_key;
+      SIValue prop_vals[2];
+
+      // Limiter for random values
+      int denom = RAND_MAX / 20 + 1;
+
+      for (int i = 0; i < n; i ++) {
+        node = NodeIterator_Next(it);
+        node->id = i;
+        // Properties will have a random value between 1 and 20
+        int prop_val = rand() / denom + 1;
+        char str_prop[10];
+        // Build a string and numeric property for each node out of the random value
+        sprintf(str_prop, "%d", prop_val);
+        prop_vals[0] = SI_StringVal(str_prop);
+        prop_vals[1] = SI_DoubleVal(prop_val);
+
+        Node_Add_Properties(node, 2, prop_keys, prop_vals);
+
+        SIValue_Free(&prop_vals[0]);
+      }
+      NodeIterator_Free(it);
+
+      return g;
+    }
+};
+
+// Define static members of IndexTest class
+const size_t IndexTest::expected_n = 100;
+const char* IndexTest::label = "test_label";
+char* IndexTest::str_key = "string_prop";
+char* IndexTest::num_key = "num_prop";
+int IndexTest::label_id;
```

Comment:
I could not find a better way to define static class members (here and in `test_skiplist.cpp`) than this, but I might just be lacking in C++ skills. Tell me if you know a clearer method!

Reply:
Why do we need those to be static?

Reply:
We override GoogleTest's static `SetUpTestCase()` method to instantiate our one graph and two indices just once for all tests. We'll get errors like `invalid use of member ‘IndexTest::str_key’ in static member function` when trying to use non-static member variables.

I feel like there's an obvious approach to this that I'm missing!